frege/compiler/gen/java/MethodCall.fr

```diff
@@ -103,19 +109,20 @@ nativeCall g (sym@SymV {nativ = Just item, gargs}) subst aexs = newBind g bsig (
              NIExtends ->
                let evalStG :: Global -> StG a -> a
                    evalStG g st = fst $ st.run g
-                   x = do g <- getST
-                          si <- symInfo sym
-                          let name  = (head si.argSigs).rho.tau.name
-                              irsym = unJust $ g.findit name
-                              nms = mapMaybe (_.name) [ fld | x@SymD{} <- values irsym.env, fld <- x.flds ]
-                          return $ flip mapMaybe nms $ \fldnm -> do
-                              nativrsym <- g.findit $ si.retSig.rho.tau.name
-                              nativsym <- TreeMap.lookup fldnm nativrsym.env
-                              nativnm <- nativsym.nativ
-                              let nativsi = evalStG g $ symInfo nativsym
-                              fldsym <- TreeMap.lookup fldnm irsym.env
-                              pure $ wrapIRMethod g (head args) (head si.argJTs) nativsi nativnm fldnm fldsym
-               in JNewClass jrty [] (evalStG g x)
+                   x = -- IR stands for an Implementation Record
+                       let irsym = unJust $ g.findit irname where
+                             irname = (head irtau.flat).name
+                             irtau  = (head sigmas).rho.tau
+                           nms = mapMaybe (_.name) [ fld | x@SymD{} <- values irsym.env, fld <- x.flds ]
+                       in
+                       flip mapMaybe nms $ \fldnm -> do
+                           nativrsym <- g.findit $ (head rty.flat).name
+                           nativsym <- TreeMap.lookup fldnm nativrsym.env
+                           nativnm <- nativsym.nativ
+                           let nativsi = evalStG g $ symInfo nativsym
+                           fldsym <- TreeMap.lookup fldnm irsym.env
+                           pure $ wrapIRMethod g (head args) (tauJT g $ head taus) fldsym fldnm nativsi nativnm
+               in JNewClass jrty (tail args) x
```

Comment:
`head args` here is the IR itself, so it's omitted. `args` is ensured to be nonempty in the `sanity` function.